internal/statestore/redis.go

```diff
@@ -423,9 +429,23 @@ func (rb *redisBackend) FilterTickets(ctx context.Context, filters []*pb.Filter,
 		}
 	}
 
+	ttl := rb.cfg.GetDuration("redis.ignoreLists.ttl")
+	curTime := time.Now()
+	curTimeInt := curTime.UnixNano()
+	startTimeInt := curTime.Add(-ttl).UnixNano()
+
+	// Filter out tickets that are fetched but not assigned within ttl time (ms).
+	idsInIgnoreLists, err = redis.Strings(redisConn.Do("ZRANGEBYSCORE", "proposed_ticket_ids", startTimeInt, curTimeInt))
```

Comment:
Do add a TODO somewhere to indicate that we need to clean up the ol ones. Create an issue even better.

Btw, a super cheap way would be to trigger a fire and forget go-routine right here to clean up everything stale in the backend. Its not clean - but for most practical purposes will work well (you can even add that to addignoredlist). Does not need to happen in this PR.